projects/dsp-ui/src/lib/viewer/views/list-view/list-view.component.ts

```diff
@@ -0,0 +1,161 @@
+import { Component, OnInit, Input, Inject, Output, EventEmitter } from '@angular/core';
+import { ReadResourceSequence, KnoraApiConnection, CountQueryResponse, ApiResponseError } from '@dasch-swiss/dsp-js';
+import { PageEvent } from '@angular/material/paginator';
+import { DspApiConnectionToken } from '../../../core';
+
+export interface fulltextSearchParams {
+    /**
+     * Iri of resource class the fulltext search is restricted to, if any.
+     */
+    limitToResourceClass?: string;
+    /**
+     * Iri of the project the fulltext search is restricted to, if any.
+     */
+    limitToProject?: string;
+    /**
+     * Iri of standoff class the fulltext search is restricted to, if any.
+     */
+    limitToStandoffClass?: string;
+}
+
+/**
+ * query: search query. It can be gravserch query or fulltext string query.
+ * The query value is expected to have at least length of 3 characters.
+ *
+ * mode: search mode "fulltext" OR "gravsearch"
+ *
+ * filter: Optional fulltext search parameter with following (optional) properties:
+ *   - limitToResourceClass: string; Iri of resource class the fulltext search is restricted to, if any.
+ *   - limitToProject: string; Iri of the project the fulltext search is restricted to, if any.
+ *   - limitToStandoffClass: string; Iri of standoff class the fulltext search is restricted to, if any.
+ */
+export interface SearchParams {
+    query: string;
+    mode: 'fulltext' | 'gravsearch';
+    filter?: fulltextSearchParams;
+}
+
+@Component({
+    selector: 'dsp-list-view',
+    templateUrl: './list-view.component.html',
+    styleUrls: ['./list-view.component.scss']
+})
+export class ListViewComponent implements OnInit {
+
+    @Input() search: SearchParams;
+
+    @Input() view?: 'list' | 'grid' = 'list';    // TODO: will be expanded with 'table' as soon as resource-table component is done
+
+    /**
+     * Click on an item will emit the resource iri
+     *
+     * @param {EventEmitter<string>} resourceSelected
+     */
+    @Output() resourceSelected: EventEmitter<string> = new EventEmitter<string>();
+
+    resources: ReadResourceSequence;
+
+    // MatPaginator Output
+    pageEvent: PageEvent;
+
+    // Number of all results
+    numberOfAllResults: number;
+
+    // in case of an api request error
+    errorMessage: ApiResponseError;
+
+    // progress status
+    loading = true;
+
+    constructor(
+        @Inject(DspApiConnectionToken) private _dspApiConnection: KnoraApiConnection
+    ) { }
+
+    ngOnInit(): void {
+        this.pageEvent = new PageEvent();
+        this.pageEvent.pageIndex = 0;
+        this._doSearch();
+    }
+
+    /**
+     *
+     * @param view 'list' | ' grid'; TODO: will be expanded with 'table' as soon as resource-table component is done
+     */
+    toggleView(view: 'list' | 'grid') {
+        this.view = view;
+    }
+
+    emitSelectedResource(id: string) {
+        this.resourceSelected.emit(id);
+    }
+
+    loadMoreResults(page: PageEvent) {
```

Comment:
@benjamingeer  implemented the boolean knora-api:mayHaveMoreResults that we may use in this component to handle more results ([Discuss discussion](https://discuss.dasch.swiss/t/knora-api-js-lib-pr-changing-public-api/158/5))

Reply:
How is this helpful in MatPaginator or how should we use it there? I mean `mayHaveMoreResults` is a boolean value and why not using the offset? It's a number that can be used in matPaginator. I do not understand 🤔

Reply:
Done in bf28aad and 6990fdc